src/Plotting/plot_recipes.jl

```diff
@@ -11,6 +11,105 @@ DEFAULT_GRID = true
 DEFAULT_ASPECT_RATIO = :none
 PLOT_PRECISION = 1e-3
 PLOT_POLAR_DIRECTIONS = 40
+DEFAULT_PLOT_LIMIT = 1000
+
+function _extract_limits(p::RecipesBase.AbstractPlot)
+    lims = Dict()
+    if length(p) > 0
+        subplot = p[1]
+        for symbol in [:x, :y]
+            lims[symbol] = subplot[Symbol(symbol,:axis)][:lims]
+        end
+    else
+        lims[:x] = :auto
+        lims[:y] = :auto
+    end
+    return lims
+end
+
+function _extract_extrema(p::RecipesBase.AbstractPlot)
+    extrema = Dict()
+    if length(p) > 0
+        subplot = p[1]
+        for symbol in [:x, :y]
+            emin = subplot[Symbol(symbol,:axis)][:extrema].emin
+            emax = subplot[Symbol(symbol,:axis)][:extrema].emax
+            extrema[symbol] = (emin, emax)
+        end
+    else
+        extrema[:x] = (-Inf, Inf)
+        extrema[:y] = (-Inf, Inf)
+    end
+    return extrema
+end
+
+function _update_plot_limits!(lims, X::LazySet)
+    box = box_approximation(X)
+    box_min = low(box)
+    box_max = high(box)
+    for (idx,symbol) in enumerate([:x, :y])
+        if lims[symbol] != :auto
+            # scaling factor for beautification
+            ϵ = 0.05
+            # compute the size of the plotting windows in the direction `symbol`
+            offset = max(lims[symbol][2], box_max[idx]) -
```

Comment:
I think `offset` is not the right term. You compute the interval approximation of two intervals (namely the current plotting range and the box approximation `X` in dimension `idx`). Ignoring the `ϵ` factor, this interval will be the new plotting range in dimension `idx`. So maybe `new_range` or `width`?